third_party/move/move-compiler-v2/src/pipeline/control_flow_graph_simplifier.rs

```diff
@@ -0,0 +1,491 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Simplifies the control flow graph by
+//! (Implemented by `ControlFlowGraphSimplifier`)
+//! - eliminating branch/jump to jump
+//!     L1: goto L2 (L1 != L2)
+//!     replacing all goto L1 by goto L2, and deleting L1: goto L2
+//! - removes edges where the source has only one successors and the target has only one predecessors
+//!     BB1: ...
+//!          goto L2
+//!     BB2: L2 (no other goto L2)
+//!          ...
+//!     =>
+//!     BB1: BB1 content
+//!          BB2 content
+//!          goto L2
+//! - removes all jumps to the next instruction:
+//!     goto L;
+//!     L;
+//! - replaces branch to same target by jump
+//!     if ... goto L else goto L
+//!     =>
+//!     goto L
+//! (Implemented by `UnreachableCodeElimination`)
+//! - removes unreachable codes
+//! (TODO: eliminate jump to branch)
+
+use move_model::model::FunctionEnv;
+use move_stackless_bytecode::{
+    function_target::FunctionData,
+    function_target_pipeline::{FunctionTargetProcessor, FunctionTargetsHolder},
+    stackless_bytecode::{Bytecode, Label},
+    stackless_control_flow_graph::{BlockId, StacklessControlFlowGraph},
+};
+use std::collections::BTreeMap;
+
+pub struct ControlFlowGraphSimplifier {}
+
+impl FunctionTargetProcessor for ControlFlowGraphSimplifier {
+    fn process(
+        &self,
+        _targets: &mut FunctionTargetsHolder,
+        fun_env: &FunctionEnv,
+        data: FunctionData,
+        _scc_opt: Option<&[FunctionEnv]>,
+    ) -> FunctionData {
+        if fun_env.is_native() {
+            return data;
+        }
+        let mut transformer = ControlFlowGraphSimplifierTransformation::new(data);
+        transformer.transform();
```

Comment:
Note that when you transform, all annotations have to be cleared (because code offsets are changed). We have a refactoring issue to consistently do this: https://github.com/aptos-labs/aptos-core/issues/11690, but until that refactor, we need to manually remember to do this.

Reply:
done